praxis-isncsci/ui
# Add random exam generation for Connect team to do testing

Closes #100 

## Problem

The **Praxis Connect** team wants to test that communication through the external message provider is working and that they can successfully save the data coming from this app. In order to do so, they need complete exams being sent from the app.

We are far behind in the interface work to allow them to manually complete a form.

## Solution

When the user presses the `Calculate` button, the system is generating a random `examData` object. We implemented most of the `calculate.useCase` so we run the random exam through the calculator to generate actual totals.

The exam data with the totals are being passed back to the hosting application through the external message provider for them to test communication and saving.

## Test

- [x] 1. Can get a random exam and totals sent through the external message provider

<details>
  <summary>Test Case</summary>

  1. Navigate to the `ExternalMessagePortProvider` **Storybook** story: `?path=/story/app-externalmessageportprovider--primary`
  2. Open the developer console on the browser
  3. Press the `Calculate` button
  4. Confirm that a random exam data is available on the interface and the same information is being passed by checking the message on the console.

</details>
